haas/api.py

```diff
@@ -183,29 +185,42 @@ def project_remove_user(project, user):
                             #############
 
 
-@rest_call('PUT', '/node/<node>')
-def node_register(node, ipmi_host, ipmi_user, ipmi_pass):
+@rest_call('PUT', '/node/<node>', schema=Schema({
+    'obm':{
+	'type': basestring, 
+	Optional(object):object,
+	},
+}))
+def node_register(node, **kwargs):
     """Create node.
 
     If the node already exists, a DuplicateError will be raised.
     """
     _assert_absent(model.Node, node)
-    node = model.Node(node, ipmi_host, ipmi_user, ipmi_pass)
-    local.db.add(node)
+    obm_type = kwargs['obm']['type']
+    cls = concrete_class_for(model.Obm, obm_type)
+    if cls is None:
+        raise BadArgumentError('%r is not a valid OBM type.' % obm_type)
+    cls.validate(kwargs['obm'])
+    node_obj = model.Node(label=node, obm=cls(**kwargs['obm']))
+#    node = model.Node(node, ipmi_host, ipmi_user, ipmi_pass)
+    local.db.add(node_obj)
     local.db.commit()
 
 
 @rest_call('POST', '/node/<node>/power_cycle')
 def node_power_cycle(node):
     node = _must_find(model.Node, node)
-    node.power_cycle()
+#    node.power_cycle()
+    node.obm.power_cycle()
 
 
 @rest_call('POST', '/node/<node>/power_off')
 def node_power_off(node):
-    db = model.Session()
-    node = _must_find(db, model.Node, node)
-    node.power_off()
+#    db = model.Session()
+#    node = _must_find(db, model.Node, node)
```

Comment:
Okay, not going to point out all of these; please go through and clean these up.